### PR TITLE
Removed useless variables for return statements

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Remove unused variables.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php:179]: `Mage::helper('core')->jsonEncode(isset($rootArray['children'])` is always `true` since `Mage::helper('core')->jsonEncode(true)` and `Mage::helper('core')->jsonEncode(false)`  return string 'true' and string 'false' when I checked. If this is correct, then the check is useless and can be just `return $rootArray['children']`.

Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php:179]: Or `return $rootArray['children'] ?? [];`

**Comment by tmotyl** [comment]
Hi @sreichel 
what is the reasoning for this change? It's  easier for debugging when the variable is there (e.g. stepping with xdebug). Does it bring any value?

File: app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php:179]: Why remove `Mage::helper('core')->jsonEncode()`? Please check closing bracers for `jsonEncode()` ... its

`return Mage::helper('core')->jsonEncode($rootArray['children'] ?? array());`

**Comment by sreichel** [comment]
@tmotyl just some cleanup. It's used ony in a very few places, everywhere else is no varaiable set.

File: app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php
Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php:179]: My bad...I missed the braces.

**Comment by ADDISON74** [comment]
Why this commit was closed? It should be merged in OpenMage.

**Review by colinmollenhour — DISMISSED** [review]
This is a very subjective change but I think Sven's goal is to get the code to meet a CS standard so that CS can be used in the future.

**Review by kiatng — APPROVED** [review]

**Comment by fballiano** [comment]
I don't know, it's a very big commit on a subjecting matter. Yes, I like the "new" code more, at the same time I wouldn't have modified 170 files just for a bit cleaner code.

Anyway this PR doesn't do any harm so I think it's ok to merge it.

@Flyingmana what do you think?

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Shipment/Packaging.php
Comment (luigifab) [inline app/code/core/Mage/Adminhtml/Block/Sales/Order/Shipment/Packaging.php:173]: can I suggest to write: `return empty($confirmationTypes[$code]) ? '' : $confirmationTypes[$code];`

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Helper/Form/Gallery.php
Comment (luigifab) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Product/Helper/Form/Gallery.php:40]: can I suggest to replace the comment by `//return $this->getContentHtml().$this->getAfterElementHtml();`

File: app/code/core/Mage/Oauth/Block/Adminhtml/Oauth/AuthorizedTokens/Grid.php
Comment (luigifab) [inline app/code/core/Mage/Oauth/Block/Adminhtml/Oauth/AuthorizedTokens/Grid.php:207]: can I suggest to replace by `return $row->getCustomerId() ?? $row->getAdminId();`

**Review by fballiano — APPROVED** [review]

**Comment by fballiano** [comment]
@sreichel could you resolve the conflicts so we can merge this PR?

@luigifab would it be ok with you if we merge this PR and you open a new one to fix your notes (which are correct)?

**Comment by tmotyl** [comment]
Please dont do this. This makes debugging harder and doesnt solve any issue (this cs rule doesnt make code easier to read in most cases).
Lets focus on having phpstan related patches merged, or introducing cs rules which brings consisteny and makes code easier to read.

**Comment by fballiano** [comment]
IMHO @tmotyl and @sreichel should decide what to do on this one, also there are some conflicts to be fixed now

**Comment by luigifab** [comment]
@fballiano yes I can make a second PR.

**Comment by ADDISON74** [comment]
I agree with @tmotyl that this PR does not solve anything but just removes the declaration of a variable from the source code, instead of returning the variable as it was originally returns that value directly. Now opinions can be divided some prefer to see return with a declared variable for easy visual identification, others prefer to put everything back.

**Comment by luigifab** [comment]
One of the inspection of PHPStorm agree with this changes 😸.

**Comment by fballiano** [comment]
> One of the inspection of PHPStorm agree with these changes 😸.

how do you use that? curious to learn

**Comment by tmotyl** [comment]
@luigifab which rule? 
I see much bigger benefit in proceeding with changes like: https://github.com/OpenMage/magento-lts/pull/2055 (I have few more changes waiting for this PR to be merged) before changes like removing variables in return statements. 
The change I linked is improving quality of work of developers a lot (less confusion on return types, better suggestions in IDE, less type related bugs.

**Comment by luigifab** [comment]
Yes, this is probably better an automated tool, but I don't understand how it works.

This is: Unnecessary local variable (example with Nexcessnet/Turpentine module):
image

**Comment by tmotyl** [comment]
ah sorry @luigifab you wrote phpStorm, and I read phpstan;) yea this rule is quite annoying in phpstorm, fortunately is easy to disable in the phpstorm config

**Comment by fballiano** [comment]
This PR have a lot of comments to implement and some conflicts that I'm not sure about, can somebody please take a look at this?

**Comment by sreichel** [comment]
> @sreichel could you resolve the conflicts so we can merge this PR?

As soon my phpStorm licenese is renewed. Hope next week ...

**Review by fballiano — CHANGES_REQUESTED** [review]
This PR have a lot of comments to implement and some conflicts to fix.

File: app/code/core/Mage/Paypal/Model/Payflow/Request.php
Comment (colinmollenhour) [inline app/code/core/Mage/Paypal/Model/Payflow/Request.php:54]: In these cases it looks like this was done for the purpose of being able to uncomment the Varien_Profiler. IMHO these lines should either be uncommented and the variable assignment kept as-is or the lines should be removed.

File: app/code/core/Mage/Oauth/Block/Adminhtml/Oauth/AuthorizedTokens/Grid.php
Comment (colinmollenhour) [reply app/code/core/Mage/Oauth/Block/Adminhtml/Oauth/AuthorizedTokens/Grid.php:186]: Wow, a double-useless variable! 😆

**Review by colinmollenhour — DISMISSED** [review]
Inspected all files..
Approve for expediency but there are some minor improvements that could be made.

**Comment by sreichel** [comment]
There are a lot. Would fix them in a separate PR.

**Comment by sreichel** [comment]
> I see much bigger benefit in proceeding with changes like: https://github.com/OpenMage/magento-lts/pull/2055 (I have few more changes waiting for this PR to be merged) before changes like removing variables in return statements.
The change I linked is improving quality of work of developers a lot (less confusion on return types, better suggestions in IDE, less type related bugs.

@tmotyl i highly appreciate your work and its defenitly more important then removing useless returned variabless, but ... this PHPStan sh** would not run, if i had not spent a few hours/days/weeks to add thousends of DOCblocks ...

It's a cosmetic change, but why not have better/cleaner code?

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Shipment/Packaging.php
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Shipment/Packaging.php:151]: Something for next (incoming) PR.

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Shipment/Packaging.php
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Shipment/Packaging.php:169]: Will fixed in separate PR.

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Helper/Form/Gallery.php
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Helper/Form/Gallery.php:34]: Separate PR if not already fixed.

File: app/code/core/Mage/Paypal/Model/Payflow/Request.php
Comment (sreichel) [reply app/code/core/Mage/Paypal/Model/Payflow/Request.php:49]: Removed these lines.

File: app/code/core/Mage/CatalogIndex/Model/Indexer/Eav.php
Comment (sreichel) [reply app/code/core/Mage/CatalogIndex/Model/Indexer/Eav.php]: Fixed.

**Review by kiatng — APPROVED** [review]

**Review by fballiano — APPROVED** [review]
let's go

**Pull request merged**
